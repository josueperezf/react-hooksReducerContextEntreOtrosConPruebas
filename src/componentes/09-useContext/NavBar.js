import React from 'react'
import {
    NavLink
  } from 'react-router-dom';
// Link y NavLink hacen lo mismo, con la diferencia de que en el NavLink se puede especificar una clase css cuando el router este en esa ruta, si no esta en esa ruta no la pone
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <NavLink exact activeClassName='active' to='./' className="navbar-brand" >Navbar</NavLink>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink exact activeClassName='active' to='./' className="nav-link" aria-current="page" >Home</NavLink>
                        <NavLink exact activeClassName='active' to='./about' className="nav-link" >About</NavLink>
                        <NavLink exact activeClassName='active' to='./login' className="nav-link" >Login</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    )
}


/*
<nav>
    <ul>
        <li>
            <Link to='./'>Home</Link>
        </li>
        <li>
            <Link to='./about'> About </Link>
        </li>
        <li>
            <Link to='./login'>Login</Link>
        </li>
    </ul>
</nav>
*/