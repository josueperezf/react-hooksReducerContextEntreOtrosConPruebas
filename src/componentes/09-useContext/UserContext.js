import { createContext } from 'react'
/**
 * UserContext permite compartir informacion entre componentes, 
 * para este ejemplo, requireo este componentes en MainApp, debe de colocarse al final .Provider, 
 * la propiedad value que alli indiquemos estara disponible entre todos los componentes q lo compartan
 * para obtener su informacion se usa el hook seria, useContext(mi-user-context)
 */
export const UserContext = createContext(null);