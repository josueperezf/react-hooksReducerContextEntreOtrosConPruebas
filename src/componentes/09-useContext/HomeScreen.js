import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
    /**
     * UserContext es un archivo que cree, al cual le defini su favores en el mainApp, y los mismo estan disponibles en cualquiera de sus componentes
     */
    const {user} = useContext(UserContext);
    console.log(user);
    return (
        <div>
            <h1>HomeScreen</h1>
            <hr/>
            <pre>
                { JSON.stringify(user, null, 4) }
            </pre>
        </div>
    )
}
