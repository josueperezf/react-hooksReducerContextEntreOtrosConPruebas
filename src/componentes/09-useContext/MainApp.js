import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    /**
     * UserContext es como un componente, al usarlo, por ser un useContext debe tener el .Provider
     * UserContext me permite pasar informacion entre componentes
    */
   const [user, setUser] = useState({});
   
    return (
        <UserContext.Provider value={{user, setUser}}>
            <AppRouter/>
        </UserContext.Provider>
    )
}
