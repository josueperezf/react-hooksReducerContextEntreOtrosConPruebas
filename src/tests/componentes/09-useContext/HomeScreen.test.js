//import { mount, shallow } from "enzyme";
import { mount } from "enzyme";
import { HomeScreen } from "../../../componentes/09-useContext/HomeScreen";
import { UserContext } from "../../../componentes/09-useContext/UserContext";

describe('Pruebas en <HomeScreen/>',()=>{
    const usuario = {
        name:'Josue',
        email:'josueperezf'
    };
    // como el homeScreen usa un useContext entonces debemos de colocarlo aqui tambien
    // en la siguiente linea se debe renderizar con mount y no con shallow ya que este solo renderiza pocos hijos de la etiqueta
    const wrapper = mount(
                        <UserContext.Provider value={usuario}>
                            <HomeScreen />
                        </UserContext.Provider>
                    );

    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();    
    })
    test('should', () => {
        
    })
    
    
});