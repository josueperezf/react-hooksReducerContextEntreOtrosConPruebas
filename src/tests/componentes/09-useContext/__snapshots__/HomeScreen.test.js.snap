// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre />
</div>
`;
